src/lib/loaders/ConfigLoader.ts

```diff
@@ -0,0 +1,15 @@
+import {CodeAnalyzerConfig} from '@salesforce/code-analyzer-core';
+
+export interface ConfigLoader {
```

Comment:
Just a stylistic thing... since I've never heard of a Loader Pattern. This essentially is the Factory pattern and this loaders look like a Factory to me.

So consider:

    export interface CodeAnalyzerConfigFactory {
        create(configFile?: string): CodeAnalyzerConfig
    }